tests/asynch/basic.rs

```diff
@@ -96,6 +96,43 @@ fn async_spawn_chain() {
     );
 }
 
+#[test]
+fn async_thread_yield() {
+    // This tests if thread::yield_now can be called from within an async block
+    check_dfs(
+        || {
+            asynch::spawn(async move {
+                thread::yield_now();
+            });
+            asynch::spawn(async move {});
+        },
+        None,
+    )
+}
+
+#[test]
+#[should_panic(expected = "assertion failed: `(left == right)`")]
```

Comment:
This assumes the Rust assertion message will never change. It might be better to add a custom message to the `assert_eq!` and check for that.